simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp

```diff
@@ -240,6 +240,21 @@ class EntityBase
 
   /*   */ auto updateTraveledDistance(const double step_time) -> double;
 
+  /*   */ auto getDistanceToTargetLaneletPose(const CanonicalizedLaneletPose & target_lanelet_pose)
+    -> double;
+
+  /*   */ bool reachPosition(
+    const geometry_msgs::msg::Pose & target_pose, const double tolerance) const;
+
+  /*   */ bool reachPosition(
+    const CanonicalizedLaneletPose & lanelet_pose, const double tolerance) const;
+
+  /*   */ bool reachPosition(const std::string & target_name, const double tolerance) const;
```

Comment:
@curious-jp Could you merge [this branch?](https://github.com/tier4/scenario_simulator_v2/pull/1233)
This branch will be merges soon and change distance measurement function as free function.